File: app/src/main/java/com/pkfcooperparry/template/ui/custom/CustomDialog.java
package com.pkfcooperparry.template.ui.custom;

import android.app.Activity;
import android.app.Dialog;
import android.view.Gravity;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.Button;
import android.widget.TextView;

import com.pkfcooperparry.template.R;

import java.util.Objects;

public class CustomDialog {
    public static final int ONE_BUTTON_DIALOG_STYLE = 0;
    public static final int TWO_BUTTON_DIALOG_STYLE = 1;

    DialogInteractionListener dialogInteractionListener;

    public void showDialog(Activity activity, DialogInteractionListener dialogInteractionListener, int type, String title, String msg) {
        setDialogInteractionListener(dialogInteractionListener);
        showDialog(activity, type, title, msg);

    }

    public void showDialog(Activity activity, int type, String title, String msg) {

        final Dialog dialog = new Dialog(activity);
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
        dialog.setCancelable(false);

        dialog.setContentView(R.layout.view_custom_dialog);
        Window window = dialog.getWindow();
        Objects.requireNonNull(window).setLayout(WindowManager.LayoutParams.WRAP_CONTENT, WindowManager.LayoutParams.WRAP_CONTENT);
        window.setBackgroundDrawableResource(android.R.color.transparent);
        window.setGravity(Gravity.CENTER);
        TextView txtTitle = dialog.findViewById(R.id.title);
        txtTitle.setText(title);

        TextView subtitle = dialog.findViewById(R.id.subtitle);
        subtitle.setText(msg);

        Button btnContinue = dialog.findViewById(R.id.btnContinue);
        Button btnCancel = dialog.findViewById(R.id.btnCancel);

        if (type == ONE_BUTTON_DIALOG_STYLE) {
            btnCancel.setVisibility(View.GONE);
            btnContinue.setText(activity.getString(R.string.okay));
        }

        btnContinue.setOnClickListener(v -> {
            if (dialogInteractionListener != null) {
                dialogInteractionListener.onContinue();
            } else
                dialog.dismiss();
        });

        btnCancel.setOnClickListener(v -> {
            if (dialogInteractionListener != null) {
                dialogInteractionListener.onCancel();
            } else
                dialog.dismiss();
        });

        dialog.show();

    }

    public DialogInteractionListener getDialogInteractionListener() {
        return dialogInteractionListener;
    }

    public void setDialogInteractionListener(DialogInteractionListener dialogInteractionListener) {
        this.dialogInteractionListener = dialogInteractionListener;
    }

    public interface DialogInteractionListener {
        void onCancel();

        void onContinue();

    }

}

File: app/src/main/java/com/pkfcooperparry/template/ui/custom/CustomDialog.kt
package com.pkfcooperparry.template.ui.custom

import android.app.Activity
import android.app.Dialog
import android.view.Gravity
import android.view.View
import android.view.Window
import android.view.WindowManager
import android.widget.Button
import android.widget.TextView

import com.pkfcooperparry.template.R

import java.util.Objects

class CustomDialog {

    var dialogInteractionListener: DialogInteractionListener? = null

    fun showDialog(activity: Activity, dialogInteractionListener: DialogInteractionListener, type: Int, title: String, msg: String) {
        this.dialogInteractionListener = dialogInteractionListener
        showDialog(activity, type, title, msg)

    }

    fun showDialog(activity: Activity, type: Int, title: String, msg: String) {

        val dialog = Dialog(activity)
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
        dialog.setCancelable(false)

        dialog.setContentView(R.layout.view_custom_dialog)
        val window = dialog.window
        Objects.requireNonNull(window).setLayout(WindowManager.LayoutParams.WRAP_CONTENT, WindowManager.LayoutParams.WRAP_CONTENT)
        window!!.setBackgroundDrawableResource(android.R.color.transparent)
        window.setGravity(Gravity.CENTER)
        val txtTitle = dialog.findViewById<TextView>(R.id.title)
        txtTitle.text = title

        val subtitle = dialog.findViewById<TextView>(R.id.subtitle)
        subtitle.text = msg

        val btnContinue = dialog.findViewById<Button>(R.id.btnContinue)
        val btnCancel = dialog.findViewById<Button>(R.id.btnCancel)

        if (type == ONE_BUTTON_DIALOG_STYLE) {
            btnCancel.visibility = View.GONE
            btnContinue.text = activity.getString(R.string.okay)
        }

        btnContinue.setOnClickListener { _ ->
            if (dialogInteractionListener != null) {
                dialogInteractionListener!!.onContinue()
            } else
                dialog.dismiss()
        }

        btnCancel.setOnClickListener { _ ->
            if (dialogInteractionListener != null) {
                dialogInteractionListener!!.onCancel()
            } else
                dialog.dismiss()
        }

        dialog.show()

    }

    interface DialogInteractionListener {
        fun onCancel()

        fun onContinue()

    }

    companion object {
        val ONE_BUTTON_DIALOG_STYLE = 0
        val TWO_BUTTON_DIALOG_STYLE = 1
    }

}
